lib/dao.js

```diff
@@ -2356,6 +2566,184 @@ DataAccessObject.prototype.unsetAttribute = function unsetAttribute(name, nullif
 };
 
 /**
+ * Replace set of attributes.
+ * Performs validation before replacing.
+ *
+ * @trigger `validation`, `save` and `update` hooks
+ * @param {Object} data Data to replace
+ * @param {Object} [options] Options for replace
+ * @param {Function} cb Callback function called with (err, instance)
+ */
+
+DataAccessObject.prototype.replace = function replace(data, options, cb) {
```

Comment:
@bajtos Sorry I'm not very clear what is the proposed approach:

Could you please confim if I understood you clearly:
You are proposing to change the change this method to the following:

Please note I added `id` to the signature of `replaceById` method and it is not a prototype method anymore.
```
DataAccessObject.replaceById = function(id, data, options, cb) {
   ...
}
```
My understanding is that still we should have a replace prototype method like the following:

```
DataAccessObject.prototype.replace = function replace(data, options, cb) {
  var Model = this.constructor;
  return Model.replaceById(getIdValue(Model, data), data, options, cb);
};
```

Did I understand it correctly?